### ISSUE TITLE
DataTable - left/right/inner/full joins

### ISSUE THREAD

**Issue by Kiyoshika** [Kiyoshika/DataTable#55]
Implement SQL-like joins on tables for left/right/inner/full 

EDIT: will need to support custom hashes for defining join conditions on specific columns (which MAY NOT be in the same order between both tables).

Can introduce a new function in HashMap such as:
```c
// will only compute the hash on specified columns
void
hash_set_columns(
    struct HashTable* const htable,
    const char (*column_names)[MAX_COL_LEN],
    const size_t n_columns);
```

Then within the hash function, will need to use the internal function `__get_multiple_column_indices`.

**Comment by Kiyoshika** [comment]
After a short break, coming back to this issue finally

**Comment by Kiyoshika** [comment]
implemented in eb34b511037d450b929b019b54bb0cb345efd09b
